contracts/contracts/coordination/Coordinator.sol

```diff
@@ -424,26 +431,103 @@ contract Coordinator is Initializable, AccessControlDefaultAdminRulesUpgradeable
-            if (participant.provider == provider) {
-                return participant;
+        if (length == 0) {
+            return (false, 0, __sentinelParticipant);
+        }
+        uint256 low = 0;
+        uint256 high = length - 1;
+        while (low <= high) {
+            uint256 mid = (low + high) / 2;
+            Participant storage middleParticipant = ritual.participant[mid];
+            if (middleParticipant.provider == provider) {
+                return (true, mid, middleParticipant);
+            } else if (middleParticipant.provider < provider) {
+                low = mid + 1;
+            } else {
+                if (mid == 0) {
+                    // prevent underflow of unsigned int
+                    break;
+                }
+                high = mid - 1;
+            }
+        }
+        return (false, 0, __sentinelParticipant);
+    }
+
+    function getParticipant(
+        Ritual storage ritual,
+        address provider
+    ) internal view returns (Participant storage) {
+        (bool found, , Participant storage participant) = findParticipant(ritual, provider);
+        require(found, "Participant not part of ritual");
+        return participant;
+    }
+
+    function getParticipant(
+        uint32 ritualId,
+        address provider,
+        bool transcript
+    ) external view returns (Participant memory, uint256) {
+        Ritual storage ritual = rituals[ritualId];
+        (bool found, uint256 index, Participant memory participant) = findParticipant(
+            ritual,
+            provider
+        );
+        require(found, "Participant not part of ritual");
+        if (!transcript) {
+            participant.transcript = "";
+        }
+        return (participant, index);
+    }
+
+    function getParticipants(
+        uint32 ritualId,
+        uint256 startIndex,
+        uint256 maxParticipants,
+        bool includeTranscript
+    ) external view returns (Participant[] memory) {
+        Ritual storage ritual = rituals[ritualId];
+        uint256 endIndex = ritual.participant.length;
+        require(startIndex >= 0, "Invalid start index");
+        require(startIndex < endIndex, "Wrong start index");
+        if (maxParticipants != 0 && startIndex + maxParticipants < endIndex) {
+            endIndex = startIndex + maxParticipants;
+        }
+        Participant[] memory ritualParticipants = new Participant[](endIndex - startIndex);
+
+        uint256 resultIndex = 0;
+        for (uint256 i = startIndex; i < endIndex; i++) {
+            Participant memory ritualParticipant = ritual.participant[i];
+            if (!includeTranscript) {
+                ritualParticipant.transcript = "";
             }
+            ritualParticipants[resultIndex++] = ritualParticipant;
         }
-        revert("Participant not part of ritual");
+
+        return ritualParticipants;
     }
 
-    function getParticipantFromProvider(
```

Comment:
This rename will cause 7.0.4 nodes to no longer work. If we want to maintain backwards compatibility we should keep the original name.

Reply:
✔️